pinot-common/src/main/java/com/linkedin/pinot/common/request/helper/ControllerRequestBuilder.java

```diff
@@ -122,7 +124,18 @@ public static JSONObject buildCreateRealtimeTableJSON(String tableName, String s
         String timeColumnName, String timeType, String retentionTimeUnit, String retentionTimeValue, int numReplicas,
     String segmentAssignmentStrategy, JSONObject streamConfigs, String schemaName, String sortedColumn,
         List<String> invertedIndexColumns, String loadMode, boolean isHighLevel, List<String> noDictionaryColumns)
-    throws JSONException {
+      throws JSONException {
```

Comment:
Should be empty map, not `null`. This way you avoid having to add a null check. See Effective Java Item 43: Return empty arrays or collections, not nulls.